### DISCUSSION TITLE
why edges are not automatically transitioned and cannot be accessed outside their original transaction?

### DISCUSSION THREAD

**Discussion by LiJie20190102** [JanusGraph/janusgraph#4344] [Q&A]
I haven't found out why the edge cannot be automatically converted based on the code. Can you tell me which code can control it?

https://docs.janusgraph.org/basics/transactions/
![1711030340820](https://github.com/JanusGraph/janusgraph/assets/53458004/480e994e-6ccc-40ca-9bf6-bb71fb0f417d)

**Comment by li-boxuan** [comment]
Just try what happens when you don't refresh the edge
